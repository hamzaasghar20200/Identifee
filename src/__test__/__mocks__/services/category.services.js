export const GetCategoriesMock = {
  data: {
    categories: [
      {
        idcreatedAt: '2021-07-18T04:48:19.365Z',
        description: 'Identifee test 1',
        id: 1,
        title: 'Test 1',
        updatedAt: '2022-02-14T18:29:19.827Z',
      },
    ],
  },
};
