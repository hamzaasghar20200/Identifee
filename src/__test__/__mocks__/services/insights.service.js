export const dataGetDashboards = {
  count: 2,
  rows: [
    {
      id: 'dashboard_1',
      name: 'Dashboard 1',
      created_by: 'user_1',
      tenant_id: 'tenant_1',
      created_at: '2022-01-13T21:53:53.702Z',
      updated_at: '2022-01-13T21:53:53.702Z',
    },
    {
      id: 'dashboard_2',
      name: 'Dashboard 2',
      created_by: 'user_1',
      tenant_id: 'tenant_1',
      created_at: '2022-01-14T21:53:53.702Z',
      updated_at: '2022-01-14T21:53:53.702Z',
    },
  ],
};

export const dataGetReports1 = {
  count: 1,
  rows: [
    {
      report_id: 'report_id_1',
      dashboard_id: 'dashboard_id_1',
      position: 0,
      insight_report: {
        id: 'insight_report_id_1',
        name: 'Lessons started and completed',
        icon: 'local_library',
        created_by: null,
        tenant_id: '000000000000',
      },
    },
  ],
};

export const dataGetReports2 = {
  count: 1,
  rows: [
    {
      report_id: 'report_id_2',
      dashboard_id: 'dashboard_id_2',
      position: 0,
      insight_report: {
        id: 'insight_report_id_2',
        name: 'Popular lessons',
        icon: 'local_library',
        created_by: null,
        tenant_id: '000000000000',
      },
    },
  ],
};
