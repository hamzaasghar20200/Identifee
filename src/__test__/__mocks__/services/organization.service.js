export const dataGetOneOwners = {
  data: [
    {
      organization_id: '12345678-2439-4c6e-a1bd-d953a4bc6cc5',
      user_id: '1',
      user: {
        id: '1',
        first_name: 'Owner 1',
        last_name: 'Additional',
        email: 'ownerAdditional1@hellobuild.co',
        title: 'Title1',
        avatar: null,
        status: 'active',
        role: '12345678-be6c-4a6e-97cf-4deaa4c3dda7',
      },
    },
  ],
  pagination: {
    count: 1,
  },
};

export const dataGetThreeOwners = {
  data: [
    {
      organization_id: '12345678-2439-4c6e-a1bd-d953a4bc6cc5',
      user_id: '1',
      user: {
        id: '1',
        first_name: 'Owner 1',
        last_name: 'Additional',
        email: 'ownerAdditional1@hellobuild.co',
        title: 'Title1',
        avatar: null,
        status: 'active',
        role: '12345678-be6c-4a6e-97cf-4deaa4c3dda7',
      },
    },
    {
      organization_id: '12345678-2439-4c6e-a1bd-d953a4bc6cc5',
      user_id: '2',
      user: {
        id: '2',
        first_name: 'Owner 2',
        last_name: 'Additional',
        email: 'ownerAdditional2@hellobuild.co',
        title: 'Title2',
        avatar: null,
        status: 'active',
        role: '12345678-be6c-4a6e-97cf-4deaa4c3dda7',
      },
    },
    {
      organization_id: '12345678-2439-4c6e-a1bd-d953a4bc6cc5',
      user_id: '3',
      user: {
        id: '3',
        first_name: 'Owner 3',
        last_name: 'Additional',
        email: 'ownerAdditional3@hellobuild.co',
        title: 'Title3',
        avatar: null,
        status: 'active',
        role: '12345678-be6c-4a6e-97cf-4deaa4c3dda7',
      },
    },
  ],
  pagination: {
    count: 3,
  },
};
