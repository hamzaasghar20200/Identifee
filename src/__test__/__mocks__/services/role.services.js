export const GetRolesMock = {
  data: [
    {
      id: '1',
      name: 'Test aaa',
      icon: 'supervised_user_circle',
      description: 'Sdferfer',
      ip_access: null,
      enforce_tfa: false,
      module_list: null,
      collection_list: null,
      admin_access: false,
      app_access: true,
    },
  ],
};
