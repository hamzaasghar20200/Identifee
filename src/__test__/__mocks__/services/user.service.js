export const dataGetGuest = {
  data: {
    users: [
      {
        id: 'guest_1',
        first_name: 'Guest',
        last_name: '1',
        email: 'guestOne@gmail.com',
        title: 'Indentifee Company',
        status: 'active',
      },
    ],
  },
};

export const dataGetThreeGuest = {
  data: {
    users: [
      {
        id: 'guest_1',
        first_name: 'Guest',
        last_name: '1',
        email: 'guestOne@gmail.com',
        title: 'Indentifee Company',
        status: 'active',
      },
      {
        id: 'guest_2',
        first_name: 'Guest2',
        last_name: '2',
        email: 'guestTwo@gmail.com',
        title: 'Indentifee Company',
        status: 'active',
      },
      {
        id: 'guest_3',
        first_name: 'Guest3',
        last_name: '3',
        email: 'guestThree@gmail.com',
        title: 'Indentifee Company',
        status: 'active',
      },
    ],
  },
};

export const getUsers = {
  data: {
    users: [
      {
        id: 'user_id_1',
        first_name: 'First Name 1',
        last_name: 'Last Name 1',
        email: 'emailUser1@helloBuild.co',
        location: 'location 1',
        title: 'title 1',
        description: 'description 1',
        tags: null,
        avatar: null,
        status: 'active',
        role: 'role_1',
        last_access: '2021-11-29T15:37:00.558Z',
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_1',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 1Test',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: false,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_2',
        first_name: 'First Name 2',
        last_name: 'Last Name 2',
        email: 'emailUser2@helloBuild.co',
        location: 'location 2',
        title: 'title 2',
        description: 'description 2',
        tags: null,
        avatar: null,
        status: 'deactivated',
        role: 'role_2',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_2',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 2',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_3',
        first_name: 'First Name 3',
        last_name: 'Last Name 3',
        email: 'emailUser3@helloBuild.co',
        location: 'location 3',
        title: 'title 3',
        description: 'description 3',
        tags: null,
        avatar: null,
        status: 'invited',
        role: 'role_3',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_3',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 3',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_4',
        first_name: 'First Name 4',
        last_name: 'Last Name 4',
        email: 'emailUser4@helloBuild.co',
        location: 'location 4',
        title: 'title 4',
        description: 'description 4',
        tags: null,
        avatar: null,
        status: 'invited',
        role: 'role_4',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_4',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 4',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_5',
        first_name: 'First Name 5',
        last_name: 'Last Name 5',
        email: 'emailUser5@helloBuild.co',
        location: 'location 5',
        title: 'title 5',
        description: 'description 5',
        tags: null,
        avatar: null,
        status: 'invited',
        role: 'role_5',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_5',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 5',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_6',
        first_name: 'First Name 6',
        last_name: 'Last Name 6',
        email: 'emailUser6@helloBuild.co',
        location: 'location 6',
        title: 'title 6',
        description: 'description 6',
        tags: null,
        avatar: null,
        status: 'invited',
        role: 'role_6',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_6',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 6',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_7',
        first_name: 'First Name 7',
        last_name: 'Last Name 7',
        email: 'emailUser7@helloBuild.co',
        location: 'location 7',
        title: 'title 7',
        description: 'description 7',
        tags: null,
        avatar: null,
        status: 'invited',
        role: 'role_7',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_7',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 7',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_8',
        first_name: 'First Name 8',
        last_name: 'Last Name 8',
        email: 'emailUser8@helloBuild.co',
        location: 'location 8',
        title: 'title 8',
        description: 'description 8',
        tags: null,
        avatar: null,
        status: 'invited',
        role: 'role_8',
        last_access: null,
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_8',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 8',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_9',
        first_name: 'First Name 9',
        last_name: 'Last Name 9',
        email: 'emailUser9@helloBuild.co',
        location: 'location 9',
        title: 'title 9',
        description: 'description 9',
        tags: null,
        avatar: null,
        status: 'active',
        role: 'role_9',
        last_access: '2029-09-09T20:02:29.430Z',
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_9',
          name: 'Rol Marisol',
          icon: 'supervised_user_circle',
          description: 'description 9',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: false,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
      {
        id: 'user_id_10',
        first_name: 'First Name 10',
        last_name: 'Last Name 10',
        email: 'emailUser10@helloBuild.co',
        location: 'location 10',
        title: 'title 10',
        description: 'description 10',
        tags: null,
        avatar: null,
        status: 'active',
        role: 'role_10',
        last_access: '2021-08-13T04:57:49.322Z',
        last_page: null,
        phone: null,
        tenant_id: '000000000000',
        roleInfo: {
          id: 'role_id_10',
          name: 'Admin',
          icon: 'supervised_user_circle',
          description: 'description 10',
          ip_access: null,
          enforce_tfa: false,
          module_list: null,
          collection_list: null,
          admin_access: true,
          app_access: true,
          tenant_id: '000000000000',
        },
      },
    ],
    pagination: {
      limit: 10,
      page: 1,
      totalPages: 4,
      count: 33,
    },
  },
};
