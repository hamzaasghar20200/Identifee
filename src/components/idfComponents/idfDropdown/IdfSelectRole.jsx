import { useEffect, useState } from 'react';
import { FormGroup, Label } from 'reactstrap';

import roleService from '../../../services/role.service';
import { ALL_LABEL } from '../../../utils/constants';
import IdfDropdownSearch from './IdfDropdownSearch';

const IdfSelectRole = ({
  label,
  defaultValue,
  value,
  onChange,
  showAll,
  query,
  disabled,
  fieldState = {},
  validationConfig = {},
  ...restProps
}) => {
  const [roleData, setRoleData] = useState([]);
  const [charactersRequire, setCharactersRequire] = useState('');
  const [searchRole, setSearchRole] = useState({
    search: '',
  });

  useEffect(() => {
    getRoles();
  }, []);

  const getRoles = async () => {
    const searchResults = await roleService
      .GetRoles(query)
      .catch((err) => console.log(err));

    const { data } = searchResults || {};

    showAll && data?.unshift({ name: ALL_LABEL });

    setRoleData(data);
  };

  const filteredData = roleData.filter((role) =>
    role.name.toLowerCase().includes(searchRole?.search?.toLowerCase())
  );
  const fieldInFields = (item) => {
    onChange({
      target: {
        name: 'status',
        value: item,
      },
    });
  };

  const stateChange = (e) => {
    const match = e.target.value.match(/([A-Za-z])/g);
    if (match && match.length >= 2) {
      setCharactersRequire('');
      setSearchRole({
        ...searchRole,
        search: e.target.value,
      });
    } else {
      return setCharactersRequire(match?.length);
    }
  };

  return (
    <FormGroup>
      {label && <Label>{label}</Label>}
      <IdfDropdownSearch
        disabled={disabled}
        title={defaultValue || 'Search for Profile'}
        data={filteredData}
        customTitle="name"
        charactersRequire={charactersRequire}
        onHandleSelect={(_, item) => fieldInFields(item)}
        value={value?.search || ''}
        onChange={stateChange}
        fieldState={fieldState}
        validationConfig={validationConfig}
        {...restProps}
      />
    </FormGroup>
  );
};

export default IdfSelectRole;
