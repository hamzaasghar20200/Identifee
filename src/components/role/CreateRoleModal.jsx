import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import {
  Spinner,
  Form,
  FormGroup,
  Label,
  CardBody,
  CardFooter,
} from 'reactstrap';
import RightPanelModal from '../modal/RightPanelModal';

import roleService from '../../services/role.service';
import {
  initialCreateRoleState,
  CANCEL_LABEL,
  ROLE_CREATED,
  ERROR_ROLE_EXIST,
  ERROR_ROLE_REQUIRED,
  ADD_NEW_PROFILE_LABEL,
} from '../../utils/constants';
import stringConstants from '../../utils/stringConstants.json';
import routes from '../../utils/routes.json';
import Alert from '../Alert/Alert';
import AlertWrapper from '../Alert/AlertWrapper';
import { useForm } from 'react-hook-form';
import AutoComplete from '../AutoComplete';
import InputValidation from '../commons/InputValidation';
import { RIGHT_PANEL_WIDTH } from '../../utils/Utils';

const constants = stringConstants.settings.roles.add;

const CreateRoleModal = ({
  showModal,
  setShowModal,
  selectedItem,
  setSelectedItem,
  ...restProps
}) => {
  const usersProfileForm = {
    name: '',
    clone: '',
    description: '',
    admin_access: false,
    owner_access: false,
  };
  const {
    register,
    handleSubmit,
    reset,
    setValue,
    formState: { errors },
  } = useForm({
    defaultValues: usersProfileForm,
  });
  const [modified, setModified] = useState(false);
  const [roles, setRoles] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [isRedirecting, setIsRedirecting] = useState(false);
  const [errorMessage, setErrorMessage] = useState(null);
  const [successMessage, setSuccessMessage] = useState(null);
  const [createRoleData, setCreateRoleData] = useState(usersProfileForm);
  const [loading, setLoading] = useState(false);

  const [searchItem, setSearchItem] = useState({
    search: '',
  });
  const history = useHistory();

  // Handler of Modal close
  const onHandleCloseModal = () => {
    reset(
      setCreateRoleData({
        name: '',
        clone: '',
        description: '',
      })
    );
    setModified(false);
    setErrorMessage(null);
    setSuccessMessage(null);
    setCreateRoleData(initialCreateRoleState);
    setShowModal(false);
    setSelectedItem('');
  };

  // Create role service
  const createNewRole = async (newRole, cloneId) => {
    setSuccessMessage(null);
    await roleService
      .CreateRole(newRole)
      .then(async (response) => {
        const permissions = await roleService.getPermissionsByRole(cloneId);
        await roleService.updatePermissions(response.id, permissions);
        setSuccessMessage(ROLE_CREATED);
        setTimeout(() => {
          history.push(`${routes.roles}/${response.id}`);
        }, 1000);
      })
      .catch(() => {
        setErrorMessage(ERROR_ROLE_EXIST);
      });
    setShowModal(false);
    reset(
      setCreateRoleData({
        name: '',
        clone: '',
        description: '',
      })
    );
  };

  // Form basic validations
  useEffect(() => {
    modified &&
      createRoleData.name === '' &&
      setErrorMessage(ERROR_ROLE_REQUIRED);
  }, [createRoleData]);

  // Capture form input changes
  const onInputChange = (e) => {
    const { name, value } = e.target || {};
    setCreateRoleData({
      ...createRoleData,
      [name]: value,
    });
    setModified(true);
  };

  // Handler of submit
  const handleCreate = async () => {
    delete createRoleData.clone;
    const clonedRole = selectedItem || roles[0];
    setIsLoading(true);
    const newRole = {
      ...createRoleData,
      admin_access: clonedRole.admin_access,
      owner_access: clonedRole.owner_access,
    };
    setCreateRoleData(newRole);
    await createNewRole(newRole, clonedRole.id);
    setSelectedItem('');
    setIsLoading(false);
    setIsRedirecting(true);
  };

  const stateChange = (e) => {
    setSearchItem({
      ...searchItem,
      search: e.target.value,
    });
  };
  const getRoles = async (count) => {
    setLoading(true);
    const result = await roleService.GetRoles(searchItem);
    setRoles(result.data);
    setLoading(false);
  };
  useEffect(() => {
    getRoles();
  }, [searchItem]);
  useEffect(() => {
    getRoles();
  }, []);
  const fieldInFields = (item) => {
    setSelectedItem(item);
    setValue('clone', item);
  };
  return (
    <RightPanelModal
      showModal={showModal}
      setShowModal={() => setShowModal(false)}
      showOverlay={true}
      containerBgColor={'pb-0'}
      containerWidth={RIGHT_PANEL_WIDTH}
      containerPosition={'position-fixed'}
      headerBgColor="bg-gray-5"
      Title={
        <div className="d-flex py-2 align-items-center text-capitalize">
          <h3 className="mb-0">{ADD_NEW_PROFILE_LABEL}</h3>
        </div>
      }
    >
      <CardBody>
        <AlertWrapper>
          <Alert
            color="danger"
            message={errorMessage}
            setMessage={setErrorMessage}
          />
          <Alert
            color="success"
            message={successMessage}
            setMessage={setSuccessMessage}
          />
        </AlertWrapper>
        <Form>
          <FormGroup>
            <Label htmlFor="name">{constants.name}</Label>
            <InputValidation
              type="input"
              name="name"
              id="name"
              required
              value={createRoleData.name}
              register={register}
              validationConfig={{
                required: `Name is required.`,
                onChange: onInputChange,
              }}
              errors={errors}
              placeholder={constants.name}
            />
          </FormGroup>
          <FormGroup>
            <Label htmlFor="clone">{constants.clone}</Label>
            <AutoComplete
              placeholder="Search for Profile"
              data={roles}
              customKey="name"
              id={'clone'}
              loading={loading}
              onHandleSelect={(item) => fieldInFields(item)}
              selected={selectedItem ? selectedItem.name : roles[0]?.name}
              onChange={stateChange}
              showIcon
              validationConfig={{
                required: `Clone is required.`,
                onChange: fieldInFields,
              }}
              register={register}
              errors={errors}
              {...restProps}
            />
          </FormGroup>
          <FormGroup>
            <Label htmlFor="description">{constants.description}</Label>
            <textarea
              type="text"
              name="description"
              rows={4}
              id="description"
              className="form-control"
              value={createRoleData.description}
              placeholder={constants.description}
              onChange={onInputChange}
            ></textarea>
          </FormGroup>
        </Form>
      </CardBody>
      <CardFooter className="px-3 text-right">
        <button
          className="btn btn-sm btn-white"
          data-dismiss="modal"
          data-uw-styling-context="true"
          onClick={onHandleCloseModal}
        >
          {CANCEL_LABEL}
        </button>
        {isRedirecting ? (
          <>
            <Spinner className={`text-primary spinner-grow-xs`} />
            <span className={`text-primary`}>{constants.redirecting}</span>
          </>
        ) : (
          <button
            type="button"
            className="btn btn-sm btn-primary ml-lg-2"
            data-uw-styling-context="true"
            onClick={handleSubmit(handleCreate)}
          >
            {isLoading ? (
              <Spinner className="spinner-grow-xs" />
            ) : (
              <span>Save</span>
            )}
          </button>
        )}
      </CardFooter>
    </RightPanelModal>
  );
};

export default CreateRoleModal;
