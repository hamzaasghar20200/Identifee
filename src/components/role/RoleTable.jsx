import React from 'react';
import { useHistory } from 'react-router';

import Table from '../GenericTable';
import routes from '../../utils/routes.json';
import MoreActions from '../MoreActions';

const RoleTable = ({
  dataSource = [],
  selectAll,
  setSelectAll,
  selectedData,
  setSelectedData,
  paginationInfo,
  setSelectedItem,
  onPageChange,
  dataInDB,
  setShowCreateRoleModal,
  sortingTable,
  setShowModal,
  sortingOrder,
}) => {
  const history = useHistory();

  const handleEditAction = (item) => {
    setSelectedItem(item);
    setShowModal(true);
  };
  const columns = [
    {
      key: 'roleName',
      orderBy: 'name',
      component: 'Profile Name',
    },
    {
      key: 'description',
      orderBy: 'description',
      component: 'Profile Description',
    },
    {
      key: 'users_count',
      orderBy: 'users_count',
      component: 'Total Users',
    },
    {
      key: 'action',
      component: 'Action',
    },
  ];

  const data = dataSource.map((dataItem) => ({
    ...dataItem,
    dataRow: [
      {
        key: 'name',
        component: (
          <h5
            className="pl-3"
            onClick={() => {
              history.push(`${routes.roles}/${dataItem.id}`);
            }}
          >
            {dataItem.name}
          </h5>
        ),
        id: dataItem.id,
      },
      {
        key: 'description',
        component: (
          <span
            onClick={() => {
              history.push(`${routes.roles}/${dataItem.id}`);
            }}
          >
            {dataItem.description}
          </span>
        ),
      },
      {
        key: 'users_count',
        component: <span>{dataItem?.totalUsers}</span>,
      },
      {
        key: 'action',
        component: (
          <div className="d-flex align-items-center">
            <a className={`icon-hover-bg cursor-pointer`}>
              <MoreActions
                icon="more_vert"
                items={[
                  {
                    id: 'edit',
                    icon: 'task_alt',
                    name: 'Clone',
                    className: dataItem.id,
                  },
                ]}
                onHandleEdit={() => handleEditAction(dataItem)}
                toggleClassName="w-auto p-0 h-auto"
              />
            </a>
          </div>
        ),
      },
    ],
  }));

  return (
    <div className="table-responsive-md datatable-custom">
      <Table
        checkbox={false}
        columns={columns}
        data={data}
        selectedData={selectedData}
        setSelectedData={setSelectedData}
        selectAll={selectAll}
        setSelectAll={setSelectAll}
        paginationInfo={paginationInfo}
        onPageChange={onPageChange}
        clickableCell
        emptyDataText="No Profiles available yet."
        title="Profile"
        dataInDB={dataInDB}
        toggle={() => setShowCreateRoleModal(true)}
        sortingTable={sortingTable}
        sortingOrder={sortingOrder}
      />
    </div>
  );
};

export default RoleTable;
