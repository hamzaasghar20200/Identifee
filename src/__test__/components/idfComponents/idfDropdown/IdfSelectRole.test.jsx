import React from 'react';
import { render, waitFor } from '@testing-library/react';
import axios from 'axios';

import IdfSelectRole from '../../../../components/idfComponents/idfDropdown/IdfSelectRole';
import { GetRolesMock } from '../../../__mocks__/services/role.services';
import roleService from '../../../../services/role.service';

jest.mock('axios');

describe('IdfSelectRole', () => {
  const id = 'selectRoleDropdown';
  const onChange = jest.fn();
  const value = { search: '' };
  const errorsRedirectHandler = jest.fn();

  const filterTestFn = jest
    .fn()
    .mockImplementation((role) => role.name.toLowerCase().includes('admin'));
  filterTestFn.mockReturnValueOnce(true);

  const filteredDataMock = GetRolesMock?.data.filter((item) =>
    filterTestFn(item)
  );

  beforeEach(async () => {
    const setRoleDataMock = jest.fn();
    const roleDataMock = (roleData) => [roleData, setRoleDataMock];
    jest.spyOn(React, 'useState').mockImplementation(roleDataMock);

    axios.get.mockImplementation(() => Promise.resolve(GetRolesMock?.data));
  });

  it('Snapshot', async () => {
    await waitFor(() => expect(filteredDataMock).toEqual(GetRolesMock?.data));

    const { container } = render(
      <IdfSelectRole id={id} onChange={onChange} value={value} />
    );

    container.filteredData = filteredDataMock;
    await waitFor(() => expect(container).toMatchSnapshot());
  });

  it('Should have a request to roles', async () => {
    expect(axios.get).not.toHaveBeenCalled();

    await roleService.GetRoles().catch((err) => errorsRedirectHandler(err));

    expect(axios.get).toHaveBeenCalled();
  });

  it('Should have a filter with correct data', () => {
    GetRolesMock?.data.filter((item) => {
      const role = item.name
        .toLowerCase()
        .includes(GetRolesMock?.data[0].name.toLowerCase());

      return waitFor(() => expect(role).toBeTruthy());
    });
  });

  it('Should show a label if exist label', () => {
    const label = 'Role section';

    const { container } = render(
      <IdfSelectRole id={id} onChange={onChange} value={value} label={label} />
    );

    const labelTag = container.querySelector('label');

    expect(labelTag).toBeTruthy();
  });

  it('Should not show a label and show a dropdown only', async () => {
    const { container } = render(
      <IdfSelectRole id={id} onChange={onChange} value={value} />
    );

    const dropdownComponent = container.querySelector(
      '#dropdown-selectRoleDropdown'
    );

    await waitFor(() => expect(dropdownComponent).toBeTruthy());

    const labelTitle = container.querySelector('label');

    await waitFor(() => expect(labelTitle).toBeNull());

    expect(labelTitle).not.toBeInTheDocument();
  });

  it('Should have a button with text Search for Category', async () => {
    const { container } = render(
      <IdfSelectRole id={id} onChange={onChange} value={value} />
    );

    const dropdownComponent = container.querySelector(
      '#dropdown-selectRoleDropdown'
    );

    await waitFor(() =>
      expect(dropdownComponent).toHaveTextContent('Search for Role')
    );
  });
});
