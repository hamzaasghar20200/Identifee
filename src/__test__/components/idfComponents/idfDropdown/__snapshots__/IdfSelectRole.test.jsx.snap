// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdfSelectRole Snapshot 1`] = `
<div>
  <div
    class="form-group"
  >
    <div
      class="dropdown"
      id="dropdown-selectRoleDropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="w-100 dropdown-search false dropdown-toggle btn btn-outline-link"
        type="button"
      >
        Search for Role
      </button>
    </div>
  </div>
</div>
`;
